### DISCUSSION TITLE
Websocket keeps throwing connection failed

### DISCUSSION THREAD

**Discussion by corpooo** [enisdenjo/graphql-ws#387] [Q&A] [answered]
**Screenshot**
This is the log of the error, log from within the `shouldRetry` function:
![CleanShot 2022-08-12 at 10 12 16@2x](https://user-images.githubusercontent.com/12585256/184303558-7b9552ad-9050-4ecb-b14d-09dc4ab4b9e7.png)

**Expected Behaviour**
It should connect to the websocket, since using the [Web Socket (old protocol)](https://www.apollographql.com/docs/react/api/link/apollo-link-ws) from the Apollo Client Docs, it connects just fine.

**Actual Behaviour**
The `client.mjs:138 WebSocket connection to 'wss://<the-url-to-the-gql-server-here>' failed: ` error is thrown.

**Debug Information**
Here is the code for the websocket client:
```js
const wsLink2 = new GraphQLWsLink(
  createClient({
    url: GRAPHQL_MIDDLEWARE_URL.replace("http", "ws"),
    shouldRetry: (err) => {
      console.log("shouldRetry:", err, err?.eventPhase);
      return true;
    },
    lazy: false,
    connectionParams: getExtraHeaders,
  })
);
```
This is the `getExtraHeaders` function:
```js
const getExtraHeaders = () => ({ Authorization: AUTH, scope: IAM_SCOPE });
```

Here is the code for the Apollo Client:
```js
// The multiplexing link that selects the correct transport for the correct
// GraphQL Operation
const splitTransportLink = split(
  // IF
  isSubscription,
  // THEN
  // wsLink,
  wsLink2,
  // ELSE
  httpLink
);

const client = new ApolloClient({
  link: ApolloLink.from([splitTransportLink]),
  cache: new InMemoryCache(),
});
```

**Further Information**
Anything else you might find helpful.

**Comment by enisdenjo** [comment, marked as answer]
> It should connect to the websocket, since using the [Web Socket (old protocol)](https://www.apollographql.com/docs/react/api/link/apollo-link-ws) from the Apollo Client Docs, it connects just fine.

As [stated in the readme disclaimer](https://github.com/enisdenjo/graphql-ws#disclaimer), this library is **not** compatible with the deprecated [`subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws). 

Please make sure that your server is running `graphql-ws` too, or is implementing the [current GraphQL over WebSocket Protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md).
